lib/database/adapters/RunAdapter.js

```diff
@@ -11,6 +11,17 @@
  * or submit itself to any jurisdiction.
  */
 
+const LHC_REVOLUTION = 11245; // The LHC revolution frequency: 11.245kHz (~ c/26.7km).
+const COLLIDING_LHC_BUNCH_CROSSINGS_IN_FILL_SCHEMA_REGEX = /[A-Za-z0-9]+_[A-Za-z0-9]+_[0-9]+_([0-9]+)_.*/;
+
+/**
+ * Extract number of colliding LHC bunch crossing (for ALICE) from LHC fill schema
+ * @param {string} fillingSchema filling schema
+ * @return {number} number of colliding LHC bunch crossing
+ */
+const extractNumberOfCollidingLhcBunchCrossings = (fillingSchema) =>
+    parseInt(fillingSchema?.match(COLLIDING_LHC_BUNCH_CROSSINGS_IN_FILL_SCHEMA_REGEX)?.[1], 10);
```

Comment:
Please do it in 2 steps to make the code more readable, and throw if fillingScheme is invalid

Reply:
refactored